### ISSUE TITLE
Working with relations.

### ISSUE THREAD

**Issue by phyllisstein** [liberation-data/drivine#64]
👋 Hi there. I've been playing with Drivine as a candidate for integrating a NestJS app with Neo4j. The design strikes me as interesting, but I'm having trouble working out how relations work. I've used the Java OGM through Scala in the past, and [this example](https://neo4j.com/developer/neo4j-ogm/#ogm-code-example) strikes me as very intuitive. But I'm not sure how it would map to this architecture.

I fear I'm just missing something obvious. If not, though, I'm curious to know what you'd recommend.

**Comment by jasperblues** [comment]
Hi @phyllisstein 

Drivine is not an OGM in the traditional sense. I explain why not on https://drivine.org 

The features that it provides: 

* Fast & easy way to create Repositories (design pattern) and integrate those into services
* Declarative transactions (like with Spring Data Neo4j, etc). 
* Streaming 

There's a sample app here: https://github.com/liberation-data/drivine-inspiration and it shows some examples of using relationships. 

## Relationships

* In Drivine *you* define a query that will return a graph projection for a [use-case](https://github.com/liberation-data/drivine-inspiration/blob/master/src/movies/moviesForActor-NEO4J.cypher). You can have it live in its own file, so you can test and profile it in its own right. 
* You inject that query into a repository
* When the query is executed drivine will to light-weight mapping of the results onto a type. 

## Why This Approach

Again the details are on the website, but in quick summary: 

* Works well on high traffic sites where queries tend to be profiled. Read: https://maxdemarzi.com/2020/05/25/declarative-query-languages-are-the-iraq-war-of-computer-science/
* In many apps one set of domain entities are not suitable for all use-cases. (Especially if these entities are also anaemic).

**Comment by phyllisstein** [comment]
Thank you for talking me through this! I'd gone through the docs, and I thought I understood the pattern better than I did. I couldn't quite work out what `bind` did once the query results came back from the database. It clicked a bit better after reading your response---just a little slow on the uptake. 😬 

I did some fiddling with your sample repo to see what this would look like in practice. I'll clean it up a bit and kick over a PR that explicitly handles relationship queries in case anyone gets lost along the same road.
